proposal/09510/README.rst

```diff
@@ -0,0 +1,64 @@
+========================
+Encryption Acceleration 
+========================
+
+
+Basic Information
+========================
+Code repository: https://github.com/09510/Encryption-Acceleration
+
+Problem to solve
+=======================
+Message encryption is an important technology to protect information security. 
+However, if you want to ensure the encryption strength, the encryption/decryption takes a lot of time.
```

Comment:
Encryption is one of the rare problems that uses numerical methods but are not covered by most numerical analysis texts.  It usually does not require many techniques that are covered by the NSD course.

If choosing this topic, you will need to make additional efforts (and arguments) to relate your work to the course.

Reply:
Thanks for your reply.

I think I got the wrong point of the project. I thought that using CUDA to accelerate calculations was a Numerical method (but it seems I was wrong). As you said, it is not special except for CUDA part.

Can I change the Topic?
If I can change, I will look at the proposals of other students and propose a new proposal within these two days.

Thanks again for your reply.

Reply:
No need to hurry.  You have made a proposal and it is good enough.

CUDA is not anything special.  It is just a tool to implement numerical methods, not the numerical methods themselves.

Encryption is a useful and meaningful topic in its own application.  I was not saying it's not "special".  You do not need to pick a "special" topic either.  What I said was that it may not allow you to practice the skills I have and will introduce in NSD.  In this sense, it's not easy to get a good grade by doing an encryption project.

Of course, nothing is impossible.  As long as you can make a good connections and arguments between encryption and the numerical calculations, it may still be a feasible direction.

You are free to change the topics, any time.  If you want to spend more time on preparing for the midterm, you do not need to hurry.  Just assess your bandwidth.

You are encouraged to have a discussion with me if you are unsure about what to do.

Reply:
After serious thought, I want to change my topic to "speedup the build of a decision tree".

The operation of the decision tree/forest requires a large amount of data processing, including many times of "sort operation" and data reading.
I once hand-carved the decision tree/forest with python. There is a lot of unnecessary file copying in python.

Using pointer and reference are the basic optimizations. 
In addition, I think maybe I can use SIMD to speed up the sort operation  when building a decision tree. 
Without SIMD, because of many sort operation, it requires at least O(2^D * M * n * logn). (D is tree depth, M is feature number)

Use seletion sort with SIMD, maybe it can be done in O(D * n^2). (If I didn't misunderstand SIMD  )

The above are my current thoughts, I think I will update my new proposal later. 
I would appreciate it if you have any suggestions.